tox.ini

```diff
@@ -30,9 +22,17 @@ setenv =
     clocale: LC_ALL = C
     devdeps: PIP_EXTRA_INDEX_URL = https://pypi.anaconda.org/scipy-wheels-nightly/simple
 
+
+allowlist_externals =
```

Comment:
This seems... a little dangerous? LoL... You think we still need this after https://github.com/tox-dev/tox/pull/2614 is merged and released?

Reply:
I've cherry-picked the .tempdir and pypi_filter fixes for #14123 and those seem to suffice as temporary fix until a new tox release is out (with the same failures on Windows and 32bit).

Reply:
None of these things should be ["located outside of the tox environment"](https://tox.wiki/en/latest/config.html#allowlist_externals) so why do we need this?

Reply:
tox 4.0.2 is out, so maybe we can revert this part and try with that release. Also see https://github.com/astropy/astropy/pull/14142#issuecomment-1341821233

Reply:
Updated  to 4.0.2, but still get:

```
tests: commands[1] .tempdir> pytest --pyargs astropy /home/maxnoe/Projects/astropy/docs
tests: failed with pytest is not allowed, use allowlist_externals to allow it
``` 

locally when running `tox -e tests` and not specifying the commands in the allow list

Reply:
@gaborbernat , any advise on tox 4 compatibility here would be greatly appreciated. Thanks!

Reply:
@dhomeier , are you sure you didn't accidentally picked up #14142 on rebase? You will have to revert those pins to pick up tox 4.0.x again.

Reply:
@dhomeier , Actions kinda rebase first before running the PR, so you will pick it up regardless on the CI. I read this somewhere but cannot find the official doc now...

Reply:
Yes, I found the merge conflict on the last push – but why can I not even see what file version is really committed/run??

Reply:
> locally when running `tox -e tests` and not specifying the commands in the allow list

Looking at the configuration seems the pytest is installed via the `text` extra here https://github.com/maxnoe/astropy/blob/remove_pypi_filter/tox.ini#LL113, so it's expected that `tests` env will not have pytest 🤔 

```ini
~/git/github/astropy on  remove_pypi_filter [!]
❯ tox c -e tests -k extras
[testenv:tests]
extras =
~/git/github/astropy on  remove_pypi_filter [!]
❯ tox c -e test -k extras
[testenv:test]
extras = test
```

Reply:
So you need to rebase to current HEAD first, then undo the changes from that rebase...
At least that seems to have got it to work now.
> Successfully installed cachetools-5.2.0 chardet-5.1.0 colorama-0.4.6 distlib-0.3.6 filelock-3.8.2 packaging-22.0 platformdirs-2.6.0 pluggy-1.0.0 pyproject-api-1.2.1 tomli-2.0.1 tox-4.0.2 virtualenv-20.17.1
Run tox
7m 30s
Run python -m tox -e py310-test-cov  -- --cov-report=xml:${GITHUB_WORKSPACE}/coverage.xml  
  python -m tox -e py310-test-cov  -- --cov-report=xml:${GITHUB_WORKSPACE}/coverage.xml  
  shell: /usr/bin/sh -e {0}

Just remains what to do about `tox-pypi-filter`

Reply:
> ```
> tests: commands[1] .tempdir> pytest --pyargs astropy /home/maxnoe/Projects/astropy/docs
> tests: failed with pytest is not allowed, use allowlist_externals to allow it
> ```
>
> locally when running `tox -e tests` and not specifying the commands in the allow list

That is with `.tempdir` existing? Without it I had a failure at
```python
py39-test: exit 2 (0.00 seconds) .tempdir> pip freeze
.pkg: _exit> python /Users/derek/opt/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  py39-test: FAIL code 2 (84.00=setup[84.00]+cmd[0.00] seconds)
```
but if it exists
```python
py39-test: commands[0] .tempdir> pip freeze
astropy @ file:///Users/derek/git/astropy-black/.tox/.tmp/package/4/astropy-5.1.dev2148%2Bg1d8e7a171.d20221208.tar.gz
attrs==22.1.0
coverage==6.5.0
exceptiongroup==1.0.4
execnet==1.9.0
hypothesis==6.60.0
iniconfig==1.1.1
numpy==1.23.5
packaging==22.0
pluggy==1.0.0
psutil==5.9.4
pyerfa==2.0.0.1
pytest==7.2.0
pytest-arraydiff==0.5.0
pytest-astropy==0.10.0
pytest-astropy-header==0.2.2
pytest-cov==4.0.0
pytest-doctestplus==0.12.1
pytest-filter-subpackage==0.1.1
pytest-mock==3.10.0
pytest-openfiles==0.5.0
pytest-remotedata==0.3.3
pytest-xdist==3.1.0
PyYAML==6.0
sortedcontainers==2.4.0
tomli==2.0.1
py39-test: commands[1] .tempdir> pytest --pyargs astropy /Users/derek/git/astropy-black/docs
===================================================================== test session starts ======================================================================
```

Reply:
So, we no longer need the `.tempdir` now in the repo like this, right? If so, please remove adding that dir and squash it out. Thanks!